File: src/Pawn.java
class Pawn {
    private Point position;
    private Boolean isCaptured;

    Pawn(Point position) {
        this.position   = position;
        this.isCaptured = false;
    }

    // commands
    void moveTo(Point position) {
        this.position = position;
    }

    void capture() {
        isCaptured = true;
    }

    // accessors
    Point getPosition() {
        return position;
    }

    int getX() {
        return position.getX();
    }

    int getY() {
        return position.getY();
    }

    boolean isCaptured() {
        return isCaptured;
    }
}

File: Sources/onitama/domain/Pawn.swift
final class Pawn {
  // properties
  var position: Point
  var isCaptured = false

  // liftime
  init(position: Point) {
    self.position = position
  }

  // commands
  func move(to position: Point) {
    self.position = position
  }

  func capture() {
    self.isCaptured = true
  }
}
